cmd/record/record.go

```diff
@@ -0,0 +1,217 @@
+	recorderFlags.Uint32Var(&maxCaptureLength, "max-capture-len", 0, "Sets the maximum capture length (zero for full capture)")
+
+	recordCmd.Flags().AddFlagSet(recorderFlags)
+	recordCmd.SetUsageTemplate(template.Usage(config.ServerFlags, recorderFlags))
+
+	return recordCmd
+}
+
+func parseProto(proto string) (recorderpb.Protocol, error) {
+	switch strings.ToUpper(proto) {
+	case "TCP":
+		return recorderpb.Protocol_PROTOCOL_TCP, nil
+	case "UDP":
+		return recorderpb.Protocol_PROTOCOL_UDP, nil
+	case "ANY":
+		return recorderpb.Protocol_PROTOCOL_ANY, nil
+	default:
+		return 0, fmt.Errorf("unsupported protocol: %s", proto)
+	}
+}
+
+func parseFilters(filters []string) ([]*recorderpb.Filter, error) {
+	if len(filters) == 0 {
+		return nil, errors.New("need to specify at least one filter")
+	}
+
+	result := make([]*recorderpb.Filter, 0, len(filters))
+	for _, filter := range filters {
+		f := strings.SplitN(filter, fitlerSep, 5)
+		if len(f) != 5 {
+			return nil, fmt.Errorf("invalid 5-tuple filter: %q", f)
+		}
+
+		srcPrefix := f[0]
+		srcPort, err := strconv.ParseUint(f[1], 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("invalid source port: %s", f[1])
+		}
+		dstPrefix := f[2]
+		dstPort, err := strconv.ParseUint(f[3], 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("invalid destination port: %s", f[3])
+		}
+		proto, err := parseProto(f[4])
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, &recorderpb.Filter{
+			SourceCidr:      srcPrefix,
+			SourcePort:      uint32(srcPort),
+			DestinationCidr: dstPrefix,
+			DestinationPort: uint32(dstPort),
+			Protocol:        proto,
+		})
+	}
+
+	return result, nil
+}
+
+func fmtTime(resp *recorderpb.RecordResponse) string {
+	return resp.GetTime().AsTime().Format(time.RFC3339)
+}
+
+func fmtStats(stats *recorderpb.RecordingStatistics) string {
+	output := fmt.Sprintf("%d packets (%d bytes) written",
+		stats.GetPacketsCaptured(), stats.GetBytesCaptured())
+	if stats.GetPacketsLost() > 0 || stats.GetBytesLost() > 0 {
+		output = fmt.Sprintf("%s and %d packets (%d bytes) lost",
+			output, stats.GetPacketsLost(), stats.GetBytesLost())
+	}
+	return output
+}
+
+func runRecord(ctx context.Context, conn *grpc.ClientConn, filters []*recorderpb.Filter) error {
+	isTTY := false
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		isTTY = true
+	}
+
+	recorder, err := recorderpb.NewRecorderClient(conn).Record(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = recorder.Send(&recorderpb.RecordRequest{
+		RequestType: &recorderpb.RecordRequest_Start{
+			Start: &recorderpb.StartRecording{
+				Filesink: &recorderpb.FileSinkConfiguration{
+					FilePrefix: fileSinkPrefix,
+				},
+				Include:          filters,
+				MaxCaptureLength: maxCaptureLength,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(os.Stderr, "Started recording. Press CTRL+C to stop.\n")
```

Comment:
Could we use the command's stdout/stderr for all messages instead of `os.Stdout/os.Stderr`?

Reply:
So this is intentional, but I can change it: All server-side messages (Status, Result, Output) are sent to stdout, while messages from the CLI (`Press CTRL+C to stop` + `Stopping`) are sent to stderr.

This allows a user to pipe the metadata of the recording into a file or something without it being cluttered with ui prompts.

Reply:
Cobra commands have [`ErrOrStderr()`](https://pkg.go.dev/github.com/spf13/cobra#Command.ErrOrStderr) and [`OutOrStdout()`](https://pkg.go.dev/github.com/spf13/cobra#Command.OutOrStdout) methods which return an `io.Writer` suitable for err or out output. This is particularly useful for testing as one can then use `SetErr()`/`SetOut()` to set the outputs to some buffer or else that implements `io.Writer`, which allows capturing the command's output and validate it. As we don't have tests for this command yet, this is not too useful atm though, hence not a blocker to me.

Reply:
Makes sense, will change it when I rebase

Reply:
Unfortunately it turns out since I need to do a tty check, having a `io.Writer` for stdout is not sufficient. I'll leave it to `os.Stdout`/`os.Stderr` for now. If we ever need to unit test the output here, we can try to figure out what Cobra recommends here.